src/main/resources/data/computercraft/lua/rom/modules/main/cc/expect.lua

```diff
@@ -88,7 +88,30 @@ local function field(tbl, index, ...)
     end
 end
 
+--- Expect a number to be within a specific range.
+--
+-- @tparam number num, The value to check.
+-- @tparam min The minimum value, if nil then `-math.huge` is used.
+-- @tparam max The maximum value, if nil then `math.huge` is used.
+-- @return The given `value`.
+-- @throws If the value is outside of the allowed range.
+local function range(num, min, max)
+  expect(1, num, "number")
+  local min = expect(2, min, "number", "nil") or -math.huge
+  local max = expect(3, max, "number", "nil") or math.huge
+  if min > max then
+      error("min must be less than or equal to max)", 2)
+  end
+  
+  if num < min or num > max then
+      error(("number outside of range (expected value to be within %d and %d)"):format(min, max), 3)
```

Comment:
So our Java code also has a similar assertion, but uses an error with the following format:

https://github.com/SquidDev-CC/CC-Tweaked/blob/e10e30f82b8f06f1d4efc7e749cbc890e1f94f6c/src/main/java/dan200/computercraft/shared/peripheral/generic/methods/ArgumentHelpers.java#L40

Which is then called with something like `assertBetween(slot, 1, size, "Slot out of range (%s)")`. I wonder if we could do something similar here? So accept an optional message to prefix it with.

It might also be worth including the original value in the error message (like `expected %d to be within %d and %d` - flexible on the format :p).